server/vulnerabilities/msrc/parsed/product_test.go

```diff
@@ -431,3 +432,115 @@ func TestFullProductName(t *testing.T) {
 		}
 	})
 }
+
+func TestProductHasDisplayVersion(t *testing.T) {
+	tc := []struct {
+		name   Product
+		result bool
+	}{
+		{
+			name:   "Windows 11 for x64-based Systems",
+			result: false,
+		},
+		{
+			name:   "Windows 11 Version 22H2 for x64-based Systems",
+			result: true,
+		},
+		{
+			name:   "Windows Server 2022, 23H2 Edition (Server Core installation)",
+			result: true,
+		},
+		{
+			name:   "Windows Server 2022 (Server Core installation)",
+			result: false,
+		},
+		{
+			name:   "Windows Server 2022",
+			result: false,
+		},
+		{
+			name:   "Windows Server, version 1803  (Server Core Installation)",
+			result: true,
+		},
+	}
+
+	for _, tt := range tc {
+		require.Equal(t, tt.result, tt.name.HasDisplayVersion(), tt.name)
+	}
+}
+
+var msrcWinProducts = Products{
+	"11926": "Windows 11 for x64-based Systems",
+	"11927": "Windows 11 for ARM64-based Systems",
+	"12085": "Windows 11 Version 22H2 for ARM64-based Systems",
+	"12086": "Windows 11 Version 22H2 for x64-based Systems",
+	"12242": "Windows 11 Version 23H2 for ARM64-based Systems",
+	"12243": "Windows 11 Version 23H2 for x64-based Systems",
+	"11923": "Windows Server 2022",
+	"11924": "Windows Server 2022 (Server Core installation)",
+	"12244": "Windows Server 2022, 23H2 Edition (Server Core installation)",
+}
+
+func TestMatchesOperatingSystem(t *testing.T) {
+	ctx := context.Background()
+	tc := []struct {
+		name string
+		os   fleet.OperatingSystem
+		want string
+		err  error
+	}{
+		{
+			name: "OS with known Display Version Match x64",
+			os: fleet.OperatingSystem{
+				Name:           "Windows 11 Pro 22H2",
+				Arch:           "x86_64",
+				DisplayVersion: "22H2",
+			},
+			want: "12086",
+			err:  nil,
+		},
+		{
+			name: "OS with known Display Version Match ARM64",
+			os: fleet.OperatingSystem{
+				Name:           "Windows 11 Pro 22H2",
+				Arch:           "ARM 64-bit Processor",
```

Comment:
Do we store `ARM 64-bit Processor` in our DB?

Reply:
I pulled that from local dev using a local Parallels VM